MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -292,6 +292,9 @@ namespace WPEFramework {
                 MaintenanceManager::_instance->onMaintenanceStatusChange(MAINTENANCE_ERROR);
                 m_statusMutex.unlock();
                 LOGINFO("Maintenance is exiting as device is not connected to internet.");
+		if (UNSOLICITED_MAINTENANCE == g_maintenance_type){
```

Comment:
better to check for "!g_unsolicited_complete" as well just like the other condition